src/ddmd/expression.d

```diff
@@ -7823,6 +7823,23 @@ extern (C++) class UnaExp : Expression
         return null;
     }
 
+    final Expression incompatibleTypes()
+    {
+        if (e1.type.toBasetype() != Type.terror)
+        {
+            if (e1.op == TOKtype)
+            {
+                error("incompatible type for (%s(%s)): cannot use '%s' with types", Token.toChars(op), e1.toChars(), Token.toChars(op));
+            }
+            else
+            {
+                error("incompatible type for (%s(%s)): '%s'", Token.toChars(op), e1.toChars(), e1.type.toChars());
+            }
+            return new ErrorExp();
+        }
+        return this;
```

Comment:
This is just the same code as in [BinExp.incompatibleTypes](https://github.com/dlang/dmd/blob/21b67f01bf25066992875cd9db95b7bcd8717c58/src/ddmd/expression.d#L7922-L7939).
It returns the orignal expression when the leaf expressions (`e1` or `e2`) are already errors.
I'm not in the business of changing that, but if you think it makes more sense, just follow up with a PR.